lib/SILOptimizer/Mandatory/PerformanceDiagnostics.cpp

```diff
@@ -755,6 +755,10 @@ class PerformanceDiagnosticsPass : public SILModuleTransform {
   void run() override {
     SILModule *module = getModule();
 
+    // Skip all performance/embedded diagnostics if not in WMO mode. Building in
+    // non-WMO mode currently results in false positives.
```

Comment:
I think we should do this only for SourceKit but not in a regular build. In embedded non-wmo builds can only happen in SourceKit, anyway. But for performance annotations (in non-wmo, non-embedded mode) such errors are not false alarms.

Reply:
@bnbarham do you know a way to detect here that we're doing a SourceKit build?

Reply:
> But for performance annotations (in non-wmo, non-embedded mode) such errors are not false alarms.

I'm confused, why should they be disabled for SourceKit then?

Reply:
Because SourceKit does not compile with wmo, regardless of the build settings